src/TheThingsNetwork.cpp

```diff
@@ -321,18 +320,42 @@ int TheThingsNetwork::poll(int port, bool confirm) {
   return sendBytes(payload, 1, port, confirm);
 }
 
-void TheThingsNetwork::airTimeValue(int payloadSize) {
+int TheThingsNetwork::getInfo(String message) {
+  int i = 5;
+  int stock = 0;
+  String str;
+  while (i <= 8) {
+    str = "";
+    str.concat(F("4/"));
+    str.concat(i);
+    if (str == message)
+      return (i - 4);
+    i = i + 1;
+  }
+  i = -1;
+  while (message[++i]) {
+    message[i] >= '0' && message[i] <= '9' ? stock = (stock + (message[i] - 48)) * 10 : stock = stock;
+  }
+  stock == 0 && message == F("on") ? stock = 1 : stock = stock / 10;
+  return (stock);
+}
+
+void TheThingsNetwork::trackAirtime(int payloadSize) {
   payloadSize = 13 + payloadSize;
-  int sf = TTN_DEFAULT_SF;
-  int ps = 8;
-  int band = 125;
+  int sf = getInfo(readValue(F("radio get sf")));
+  int ps = getInfo(readValue(F("radio get prlen")));
+  int band = getInfo(readValue(F("radio get bw")));
+  int header = getInfo(readValue(F("radio get crc")));
+  int cr = getInfo(readValue(F("radio get cr")));
```

Comment:
These values do not change all the time right? I think it's more power efficient to read these values once and use the variables instead of reading them out of the RN module